flink-streaming-java/src/main/java/org/apache/flink/streaming/api/connector/sink2/WithPreWriteTopology.java

```diff
@@ -21,18 +21,9 @@
 import org.apache.flink.annotation.Experimental;
 import org.apache.flink.api.connector.sink2.Sink;
 import org.apache.flink.api.connector.sink2.SinkWriter;
-import org.apache.flink.streaming.api.datastream.DataStream;
 
 /** Allows expert users to implement a custom topology before {@link SinkWriter}. */
 @Experimental
-public interface WithPreWriteTopology<InputT> extends Sink<InputT> {
-
-    /**
-     * Adds an arbitrary topology before the writer. The topology may be used to repartition the
-     * data.
-     *
-     * @param inputDataStream the stream of input records.
-     * @return the custom topology before {@link SinkWriter}.
-     */
-    DataStream<InputT> addPreWriteTopology(DataStream<InputT> inputDataStream);
-}
+@Deprecated
```

Comment:
we could explain in the javadoc why this interface is deprecated, specially since it's a new addition

Reply:
Added the `@deprecated` in #24022